lib/salus/scanners/report_node_modules.rb

```diff
@@ -103,6 +110,36 @@ def record_dependencies_from_package_lock_json
       end
     end
 
+    def record_dependencies_from_bower_json
+      packages = JSON.parse(@repository.bower_json)
+      # Record if custom registry is defined in .bowerrc
+      # Supports single and multiple registry -
+      # {"registry": "https://registry.bower.io"}
+      # {"registry": {"search": ["http://localhost:9000","https://registry.bower.io"] }}
+      sources = '<bower manager default>'
+      if @repository.bowerrc_present?
+        file = JSON.parse(@repository.bowerrc)
+        if file.key?("registry")
+          is_nested = file["registry"].is_a? String
+          sources = is_nested ? file["registry"] : file["registry"]["search"].join(',')
```

Comment:
1. If `file["registry"]` is not a String, then is it always a hash?   If you are not sure, then maybe check if `is_a? Hash`.
2. Even if `registry` is a Hash, the `search` key could be missing, which would then lead to a `join` not defined on `nil` exception.

Reply:
Good point, let me add extra checks